app/lib/data_loader/calculator.rb

```diff
@@ -0,0 +1,69 @@
+class DataLoader::Calculator
+  def self.call(current:, future:)
+    raise ArgumentError if current.length > 2 || current.length != future.length
+
+    if current.one?
+      SpinePoint.new(current: current.first, future: future.first).call
+    else
+      PayBand.new(current: Range.new(*current), future: Range.new(*future)).call
+    end
+  end
+
+  class Base
+    attr_reader :current, :future
+
+    def initialize(current:, future:)
+      @current = current
+      @future = future
+    end
+
+    def call
+      {
+        type: type,
+        current: current,
+        future: future,
+        increase: increase,
+        increase_percentage: increase_percentage
+      }
```

Comment:
you should be able to omit the hash value for all these keys

```
      {
        type:,
        current:,
        future:,
        increase:,
        increase_percentage:
      }
```

Reply:
True! Can't get used to that syntax. I wonder why they didn't just go for JS approach here `{type, current, ...}`